sdk-lib/src/main/java/cash/z/ecc/android/sdk/SdkSynchronizer.kt

```diff
@@ -197,8 +188,11 @@ class SdkSynchronizer private constructor(
     // Transactions
     //
 
-    override val clearedTransactions get() = storage.allTransactions
-    override val pendingTransactions = txManager.getAll()
+    override val transactions
+        get() = combine(processor.networkHeight, storage.allTransactions) { networkHeight, allTransactions ->
```

Comment:
I think that, given the immutable nature of the blockchain, it's safe to use the latest height of the chain instead of the latest scanned height to settle confirmed, pending and expired. 

Also, a performance concern: how does this combination of stream work? 

When you scan you will be progressing in the last known scanned height with every scanned batch, would that cause a re-combination of these two streams every time one of them emits a new value?